File: tuan2/PhanA/bai38.py
# Python (bai38.py)
import sys

def main():
    input_data = sys.stdin.read().split()
    if not input_data: return

    n = int(input_data[0])
    a = [int(x) for x in input_data[1:n + 1]]

    strict_inc = True
    non_dec = True

    for i in range(n - 1):
        if a[i] >= a[i + 1]:
            strict_inc = False
        if a[i] > a[i + 1]:
            non_dec = False
        if not strict_inc and not non_dec:
            break

    res1 = "YES" if strict_inc else "NO"
    res2 = "YES" if non_dec else "NO"
    print(f"{res1} — {res2}")

if __name__ == "__main__":
    main()

File: tuan2/PhanA/bai38.jl
# Julia (bai38.jl)
function main()
    tokens = split(read(stdin, String))
    if isempty(tokens) return end

    n = parse(Int, tokens[1])
    a = [parse(Int, tokens[i]) for i in 2:(n + 1)]

    strict_inc = true
    non_dec = true

    for i in 1:(n - 1)
        if a[i] >= a[i + 1]
            strict_inc = false
        end
        if a[i] > a[i + 1]
            non_dec = false
        end
        if !strict_inc && !non_dec
            break
        end
    end

    res1 = strict_inc ? "YES" : "NO"
    res2 = non_dec ? "YES" : "NO"
    println("$res1 — $res2")
end

main()